src/main/java/org/kiwiproject/dropwizard/util/health/HttpConnectionsHealthCheck.java

```diff
@@ -0,0 +1,205 @@
+package org.kiwiproject.dropwizard.util.health;
+
+import static java.util.Collections.emptyList;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toMap;
+import static java.util.stream.Collectors.toSet;
+import static org.kiwiproject.base.KiwiPreconditions.checkArgument;
+import static org.kiwiproject.base.KiwiPreconditions.requireNotNull;
+import static org.kiwiproject.metrics.health.HealthCheckResults.newHealthyResult;
+import static org.kiwiproject.metrics.health.HealthCheckResults.newResultBuilder;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.health.HealthCheck;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+
+/**
+ * Health check that checks the percent of leased connections against the maximum number of connections for
+ * JAX-RS {@link javax.ws.rs.client.Client} instances that were created using Dropwizard's
+ * {@code io.dropwizard.client.JerseyClientBuilder}, which creates an HTTP connection pool and registers various
+ * connection metrics that we can query.
+ * <p>
+ * Please note, if using a default JAX-RS {@link javax.ws.rs.client.Client} created using the normal
```

Comment:
This comment made me go back into registry-aware-jersey-client...the regular `ClientBuilders.jersey()` also returns a `Client` that doesn't have any metrics attached. We do provide the `DropwizardManagedClientBuilder` which uses the Dropwizard stuff and attaches metrics. Do we care that clients built using `ClientBuilders.jersey()` don't have metrics? I guess they also don't have an HTTP connection pool, either...

Reply:
Yeah that is an issue, but hopefully we will be using the dropwizard managed ones.